### ISSUE TITLE
error for crossentropy on GPU: ERROR: Broadcast output type Any is not concrete

### ISSUE THREAD

**Issue by osofr** [FluxML/Flux.jl#201]
Running the LSTM example on GPUs with `crossentropy`  loss produces the following error. Different losses (e.g., `mse`) work fine. Snippet below, for full replication run example in #189.

```julia
julia> function losscu(xs, ys)
         l = sum(crossentropy.(predict.(xs), gpu.(ys)))
         Flux.truncate!(mcu)
         return l
       end
losscu (generic function with 1 method)

julia> predict(x) = collect(x) |> gpu |> mcu
predict (generic function with 1 method)

julia> @time Flux.train!(losscu, zip(Xs[1:1], Ys[1:1]), opt, cb = throttle(evalcb, 30))
ERROR: Broadcast output type Any is not concrete
Stacktrace:
 [1] broadcast_t at /home/ubuntu/.julia/v0.6/CuArrays/src/broadcast.jl:34 [inlined]
 [2] broadcast_c at /home/ubuntu/.julia/v0.6/CuArrays/src/broadcast.jl:63 [inlined]
 [3] broadcast at ./broadcast.jl:455 [inlined]
 [4] tracked_broadcast(::Function, ::Flux.OneHotMatrix{CuArray{Flux.OneHotVector,1}}, ::TrackedArray{…,CuArray{Float32,2}}, ::Int64) at /home/ubuntu/.julia/v0.6/Flux/src/tracker/array.jl:278
 [5] #crossentropy#71(::Int64, ::Function, ::TrackedArray{…,CuArray{Float32,2}}, ::Flux.OneHotMatrix{CuArray{Flux.OneHotVector,1}}) at /home/ubuntu/.julia/v0.6/Flux/src/layers/stateless.jl:8
 [6] crossentropy(::TrackedArray{…,CuArray{Float32,2}}, ::Flux.OneHotMatrix{CuArray{Flux.OneHotVector,1}}) at /home/ubuntu/.julia/v0.6/Flux/src/layers/stateless.jl:8
 [7] (::##61#62)(::Flux.OneHotMatrix{Array{Flux.OneHotVector,1}}, ::Flux.OneHotMatrix{Array{Flux.OneHotVector,1}}) at ./<missing>:0
 [8] broadcast_t(::Function, ::Type{Any}, ::Tuple{Base.OneTo{Int64}}, ::CartesianRange{CartesianIndex{1}}, ::Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1}, ::Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1}) at ./broadcast.jl:258
 [9] broadcast_c at ./broadcast.jl:321 [inlined]
 [10] broadcast at ./broadcast.jl:455 [inlined]
 [11] losscu(::Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1}, ::Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1}) at ./REPL[231]:2
 [12] #train!#130(::Flux.#throttled#14, ::Function, ::Function, ::Base.Iterators.Zip2{Array{Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1},1},Array{Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1},1}}, ::Flux.Optimise.##71#75) at /home/ubuntu/.julia/v0.6/Flux/src/optimise/train.jl:39
 [13] (::Flux.Optimise.#kw##train!)(::Array{Any,1}, ::Flux.Optimise.#train!, ::Function, ::Base.Iterators.Zip2{Array{Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1},1},Array{Array{Flux.OneHotMatrix{Array{Flux.OneHotVector,1}},1},1}}, ::Function) at ./<missing>:0
```

**Comment by boathit** [comment]
It is indeed a bug. I recommend you to remove the `softmax` layer in your model and use the `logitcrossentropy` loss function which is equivalent to `crossentropy(softmax(xs), ys)` but is more numerically stable.

**Comment by MikeInnes** [comment]
This should be fixed on master, but let me know there are other issues.
